web/src/utils/column-parser.ts

```diff
@@ -0,0 +1,149 @@
+import { getRecordValue, Record } from '../api/ipfix';
+import { Column, ColumnConfigDef, ColumnsId, ColValue, KubeObj } from './columns';
+import { FieldConfig, FieldType } from './fields';
+
+const getColumnOrRecordValue = (
+  columns: Column[],
+  record: Record,
+  arg: string,
+  defaultValue: string | number
+): ColValue => {
+  if (arg.startsWith(`'`) && arg.endsWith(`'`)) {
+    // literal
+    return arg.substring(1, arg.length - 1);
+  } else if (arg.startsWith('column.')) {
+    const colId = arg.replace('column.', '');
+    const found = columns.find(c => c.id === colId);
+    if (found) {
+      return found.value(record);
+    }
+    return defaultValue;
+  }
+  return getRecordValue(record, arg, defaultValue);
+};
+
+const funcs: { [name: string]: (columns: Column[], record: Record, args: string[]) => ColValue } = {
+  concat: (columns: Column[], record: Record, args: string[]): ColValue => {
+    if (args.length < 2) {
+      console.error('getDefaultColumns - invalid parameters for concat calculated value', args);
+      return '';
+    }
+    return args.map(a => getColumnOrRecordValue(columns, record, a, '')).join('');
+  },
+  kubeObject: (columns: Column[], record: Record, args: string[]): ColValue => {
+    if (args.length !== 4) {
+      console.error('getDefaultColumns - invalid parameters for kubeObject calculated value', args);
+      return '';
+    }
+    const obj: KubeObj = {
+      kind: String(getColumnOrRecordValue(columns, record, args[0], '')),
+      namespace: String(getColumnOrRecordValue(columns, record, args[1], '')),
+      name: String(getColumnOrRecordValue(columns, record, args[2], '')),
+      showNamespace: args[3] === '1'
+    };
+    if (!obj.name || !obj.kind) {
+      return undefined;
+    }
+    // Remove empty namespace
+    if (obj.namespace === '') {
+      delete obj.namespace;
+    }
+    return obj;
```

Comment:
It would feel easier to read building the KubeObj at return statement:
```suggestion
      const kind = String(getColumnOrRecordValue(columns, record, args[0], ''));
      const namespace = String(getColumnOrRecordValue(columns, record, args[1], ''));
      const name = String(getColumnOrRecordValue(columns, record, args[2], ''));
      const showNamespace = args[3] === '1'
    };
    if (!name || !kind) {
      return undefined;
    } else if (namespace === '') {
      return { kind, name };
    }
    return { kind, namespace, name, showNamespace };
```